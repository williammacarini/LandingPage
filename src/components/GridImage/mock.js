export default {
  background: false,
  title: 'My grid',
  description: 'Test my grid to verify my text here',
  grid: [
    {
      altText: 'Test 1',
      srcImg: 'assets/images/azure.svg'
    },
    {
      altText: 'Test 2',
      srcImg: 'assets/images/azure.svg'
    },
    {
      altText: 'Test 3',
      srcImg: 'assets/images/azure.svg'
    },
    {
      altText: 'Test 4',
      srcImg: 'assets/images/azure.svg'
    },
    {
      altText: 'Test 5',
      srcImg: 'assets/images/azure.svg'
    },
    {
      altText: 'Test 6',
      srcImg: 'assets/images/azure.svg'
    },
  ]
}
