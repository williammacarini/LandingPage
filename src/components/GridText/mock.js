export default {
  background: false,
  title: 'My grid',
  description: 'Test my grid to verify my text here',
  grid: [
    {
      title: 'Test 1',
      description: `Rhoncus, suscipit augue dignissim. Cum facilisi urna consectetur et mattis rhoncus dignissim tempor sed. Lorem lacinia vulputate tempor ridiculus morbi sodales nostra aliquam nulla. Massa in habitant per odio litora ornare posuere commodo. Tellus lectus aptent sit praesent est auctor euismod. Molestie semper justo interdum enim at curae; ut. Scelerisque fames nullam integer. Neque per imperdiet magnis volutpat turpis nascetur semper at pretium. Pharetra auctor tincidunt rutrum cursus. Mi euismod magna inceptos neque aenean maecenas. Mus.`,
    },
    {
      title: 'Test 2',
      description: `Rhoncus, suscipit augue dignissim. Cum facilisi urna consectetur et mattis rhoncus dignissim tempor sed. Lorem lacinia vulputate tempor ridiculus morbi sodales nostra aliquam nulla. Massa in habitant per odio litora ornare posuere commodo. Tellus lectus aptent sit praesent est auctor euismod. Molestie semper justo interdum enim at curae; ut. Scelerisque fames nullam integer. Neque per imperdiet magnis volutpat turpis nascetur semper at pretium. Pharetra auctor tincidunt rutrum cursus. Mi euismod magna inceptos neque aenean maecenas. Mus.`,
    }
  ]
}
