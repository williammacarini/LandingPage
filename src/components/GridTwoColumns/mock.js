export default
{
  title: 'Grid Two Column',
  text: `Lorem ipsum, dolor sit amet consectetur adipisicing elit.
  Eligendi consequatur asperiores minus impedit ipsa iure non labore! Corporis unde illo quos?
  Optio ex quod, saepe veniam et pariatur minus in.`,
  srcImg: 'assets/images/azure.svg'
}
