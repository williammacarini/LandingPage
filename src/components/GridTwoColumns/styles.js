import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.font.media.lteMedium} {
      grid-template-columns: 1fr;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.xlarge};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.font.media.lteMedium} {
      margin-bottom: ${theme.spacings.large};
      text-align: center;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
  `}
`;
