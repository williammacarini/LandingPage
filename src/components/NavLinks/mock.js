export default [
  {
    children: 'Link1',
    link: '#targe1',
    newTab: false,
  },
  {
    children: 'Link2',
    link: '#targe2',
    newTab: false,
  },
  {
    children: 'Link3',
    link: '#targe3',
    newTab: false,
  },
  {
    children: 'Link4',
    link: '#targe4',
    newTab: false,
  },
  {
    children: 'Link5',
    link: '#targe5',
    newTab: false,
  },
  {
    children: 'Link6',
    link: '#targe6',
    newTab: true,
  },
  {
    children: 'Link7',
    link: '#targe7',
    newTab: false,
  },
  {
    children: 'Link8',
    link: '#targe8',
    newTab: false,
  },
  {
    children: 'Link9',
    link: '#targe9',
    newTab: true,
  },
  {
    children: 'link10',
    link: 'link10',
    newTab: false,
  },
];
