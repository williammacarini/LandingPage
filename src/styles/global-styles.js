import { createGlobalStyle, css } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  * {
  }
  body {
    ${({ theme }) => css`
      background: ${theme.colors.mainBg};
    `}
  }
`;
