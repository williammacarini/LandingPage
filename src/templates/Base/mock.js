import linksMock from '../../components/NavLinks/mock';

export default {
  links: linksMock,
  logoData: {
    text: 'Logo',
    link: '#',
  },
  footerHtml: '<p>Test this Footer</p>'
}
