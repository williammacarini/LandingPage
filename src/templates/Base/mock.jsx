import React from 'react';
import { GridText } from '../../components/GridText';
import linksMock from '../../components/NavLinks/mock';
import gridMock from '../../components/GridText/mock';

export const mockBase = {
  children: (
    <>
      <GridText {...gridMock} />
      <GridText {...gridMock} background />
      <GridText {...gridMock} />
    </>
  ),
  links: linksMock,
  logoData: {
    text: 'Logo',
    link: '#',
  },
  footerHtml: '<p>Test this Footer</p>'
};
