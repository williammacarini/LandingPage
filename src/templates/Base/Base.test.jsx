import React from 'react';
import { renderTheme } from '../../styles/render-theme';
import { Base } from '.';
import { mockBase } from './mock';

describe('<Base />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Base {...mockBase}/>);
    expect(container).toMatchSnapshot();
  });
});
