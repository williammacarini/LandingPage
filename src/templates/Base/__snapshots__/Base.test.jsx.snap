// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should render 1`] = `
.c11 {
  padding-top: 5.4rem;
}

.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c7 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c14 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c19 {
  color: #0A1128;
  font-size: 3.2rem;
  text-transform: none;
}

.c21 {
  color: #FFFFFF;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c22 {
  color: #FFFFFF;
  font-size: 3.2rem;
  text-transform: none;
}

.c2 {
  position: fixed;
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  border-bottom: #DDDDDD;
  background: #FFFFFF;
  transition: all 300ms ease-in-out;
}

.c2 >.c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c2 .c6 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background: #0A1128;
  color: #FFFFFF;
  border: none;
  display: none;
  pointer-events: all;
}

.c0 >svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c8 {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
}

.c8 >img {
  max-height: 2.5rem;
}

.c9 {
  display: flex;
  flex-flow: row wrap;
}

.c10 {
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
  position: relative;
}

.c10::after {
  content: '';
  position: absolute;
  bottom: 0.76rem;
  left: 50%;
  width: 0%;
  height: 0.2rem;
  background: #dc143c;
  transition: all 300ms ease-in-out;
}

.c10:hover::after {
  left: 25%;
  width: 50%;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c16 {
  font-size: 2.4rem;
}

.c23 {
  text-align: center;
  border-top: 0.1rem solid #DDDDDD;
}

.c23 a {
  color: inherit;
  text-decoration: none;
}

.c23 .c15 {
  font-size: 1.6rem;
}

.c23 .c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c24 {
  position: fixed;
  background: #0A1128;
  color: #FFFFFF;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 4rem;
  right: 2rem;
  z-index: 6;
  opacity: 0.7;
}

.c13 .c15 {
  margin-bottom: 6.4rem;
}

.c17 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 3.2rem;
}

.c18 .c6 {
  position: relative;
  left: 5rem;
}

.c18 .c6::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 7rem;
  top: -3rem;
  left: -5rem;
  transform: rotate(5deg);
}

.c12 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c20 {
  background: #FFFFFF;
  color: #0A1128;
  background: #0A1128;
  color: #FFFFFF;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

@media (max-width: 768px) {
  .c11 {
    padding-top: 0;
  }
}

@media (max-width: 768px) {
  .c14 {
    font-size: 4rem;
  }
}

@media (max-width: 768px) {
  .c21 {
    font-size: 4rem;
  }
}

@media (max-width: 768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
  }

  .c2 >.c3 {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100vh;
    align-items: center;
    overflow-y: auto;
  }

  .c2 .c6 {
    padding-bottom: 3.2rem;
    display: flex;
    justify-content: center;
  }
}

@media (max-width: 768px) {
  .c5 {
    display: block;
    text-align: center;
    padding: 4.8rem 0;
  }
}

@media (max-width: 768px) {
  .c0 {
    display: flex;
    align-items: center;
    justify-content: center;
  }
}

@media (max-width: 768px) {
  .c9 {
    flex-flow: column wrap;
    align-content: center;
  }
}

@media (max-width: 768px) {
  .c17 {
    grid-template-columns: 1fr;
  }
}

<div>
  <button
    aria-label="Open/Close Menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open Menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6 c7"
        >
          <a
            class="c8"
            href="#"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main Menu"
          class="c9"
        >
          <a
            class="c10"
            href="#targe1"
            target="_self"
          >
            Link1
          </a>
          <a
            class="c10"
            href="#targe2"
            target="_self"
          >
            Link2
          </a>
          <a
            class="c10"
            href="#targe3"
            target="_self"
          >
            Link3
          </a>
          <a
            class="c10"
            href="#targe4"
            target="_self"
          >
            Link4
          </a>
          <a
            class="c10"
            href="#targe5"
            target="_self"
          >
            Link5
          </a>
          <a
            class="c10"
            href="#targe6"
            target="_blank"
          >
            Link6
          </a>
          <a
            class="c10"
            href="#targe7"
            target="_self"
          >
            Link7
          </a>
          <a
            class="c10"
            href="#targe8"
            target="_self"
          >
            Link8
          </a>
          <a
            class="c10"
            href="#targe9"
            target="_blank"
          >
            Link9
          </a>
          <a
            class="c10"
            href="link10"
            target="_self"
          >
            link10
          </a>
        </nav>
      </div>
    </div>
  </div>
  <div
    class="c11"
  >
    <div
      class="c12"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h2
            class="c6 c14"
          >
            My grid
          </h2>
          <div
            class="c15 c16"
          >
            Test my grid to verify my text here
          </div>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                Test 1
              </h3>
              <div
                class="c15 c16"
              >
                Rhoncus, suscipit augue dignissim. Cum facilisi urna consectetur et mattis rhoncus dignissim tempor sed. Lorem lacinia vulputate tempor ridiculus morbi sodales nostra aliquam nulla. Massa in habitant per odio litora ornare posuere commodo. Tellus lectus aptent sit praesent est auctor euismod. Molestie semper justo interdum enim at curae; ut. Scelerisque fames nullam integer. Neque per imperdiet magnis volutpat turpis nascetur semper at pretium. Pharetra auctor tincidunt rutrum cursus. Mi euismod magna inceptos neque aenean maecenas. Mus.
              </div>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                Test 2
              </h3>
              <div
                class="c15 c16"
              >
                Rhoncus, suscipit augue dignissim. Cum facilisi urna consectetur et mattis rhoncus dignissim tempor sed. Lorem lacinia vulputate tempor ridiculus morbi sodales nostra aliquam nulla. Massa in habitant per odio litora ornare posuere commodo. Tellus lectus aptent sit praesent est auctor euismod. Molestie semper justo interdum enim at curae; ut. Scelerisque fames nullam integer. Neque per imperdiet magnis volutpat turpis nascetur semper at pretium. Pharetra auctor tincidunt rutrum cursus. Mi euismod magna inceptos neque aenean maecenas. Mus.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c20"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h2
            class="c6 c21"
          >
            My grid
          </h2>
          <div
            class="c15 c16"
          >
            Test my grid to verify my text here
          </div>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                Test 1
              </h3>
              <div
                class="c15 c16"
              >
                Rhoncus, suscipit augue dignissim. Cum facilisi urna consectetur et mattis rhoncus dignissim tempor sed. Lorem lacinia vulputate tempor ridiculus morbi sodales nostra aliquam nulla. Massa in habitant per odio litora ornare posuere commodo. Tellus lectus aptent sit praesent est auctor euismod. Molestie semper justo interdum enim at curae; ut. Scelerisque fames nullam integer. Neque per imperdiet magnis volutpat turpis nascetur semper at pretium. Pharetra auctor tincidunt rutrum cursus. Mi euismod magna inceptos neque aenean maecenas. Mus.
              </div>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                Test 2
              </h3>
              <div
                class="c15 c16"
              >
                Rhoncus, suscipit augue dignissim. Cum facilisi urna consectetur et mattis rhoncus dignissim tempor sed. Lorem lacinia vulputate tempor ridiculus morbi sodales nostra aliquam nulla. Massa in habitant per odio litora ornare posuere commodo. Tellus lectus aptent sit praesent est auctor euismod. Molestie semper justo interdum enim at curae; ut. Scelerisque fames nullam integer. Neque per imperdiet magnis volutpat turpis nascetur semper at pretium. Pharetra auctor tincidunt rutrum cursus. Mi euismod magna inceptos neque aenean maecenas. Mus.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h2
            class="c6 c14"
          >
            My grid
          </h2>
          <div
            class="c15 c16"
          >
            Test my grid to verify my text here
          </div>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                Test 1
              </h3>
              <div
                class="c15 c16"
              >
                Rhoncus, suscipit augue dignissim. Cum facilisi urna consectetur et mattis rhoncus dignissim tempor sed. Lorem lacinia vulputate tempor ridiculus morbi sodales nostra aliquam nulla. Massa in habitant per odio litora ornare posuere commodo. Tellus lectus aptent sit praesent est auctor euismod. Molestie semper justo interdum enim at curae; ut. Scelerisque fames nullam integer. Neque per imperdiet magnis volutpat turpis nascetur semper at pretium. Pharetra auctor tincidunt rutrum cursus. Mi euismod magna inceptos neque aenean maecenas. Mus.
              </div>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                Test 2
              </h3>
              <div
                class="c15 c16"
              >
                Rhoncus, suscipit augue dignissim. Cum facilisi urna consectetur et mattis rhoncus dignissim tempor sed. Lorem lacinia vulputate tempor ridiculus morbi sodales nostra aliquam nulla. Massa in habitant per odio litora ornare posuere commodo. Tellus lectus aptent sit praesent est auctor euismod. Molestie semper justo interdum enim at curae; ut. Scelerisque fames nullam integer. Neque per imperdiet magnis volutpat turpis nascetur semper at pretium. Pharetra auctor tincidunt rutrum cursus. Mi euismod magna inceptos neque aenean maecenas. Mus.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="c23"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c15 c16"
        >
          <p>
            Test this Footer
          </p>
        </div>
      </div>
    </footer>
  </div>
  <a
    aria-label="Go to top"
    class="c24"
    href="#"
    title="Go to top"
  >
    <svg
      aria-hidden="true"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
      />
    </svg>
  </a>
</div>
`;
